lib/src/solver/incompatibility_cause.dart

```diff
@@ -96,6 +97,11 @@ class SdkCause extends IncompatibilityCause {
 
   @override
   String? get hint {
+    if (sdk.isDartSdk &&
+        !LanguageVersion.fromSdkConstraint(constraint).supportsNullSafety &&
+        sdk.version! >= Version(3, 0, 0).firstPreRelease) {
+      return 'The lower bound of "$constraint" does not enable null safety.';
```

Comment:
```suggestion
      return 'The lower-bound of "$constraint" does not enable null safety.';
```

Reply:
Whoops - I missed this. Will do a follow-up.

Reply:
Actually researching, I think this should be spelled in two words.

Examples:

https://en.wikipedia.org/wiki/Upper_and_lower_bounds
https://www.dictionary.com/browse/lower-bound
https://www.geeksforgeeks.org/lower_bound-in-cpp/